### PR TITLE
HPCC-20332 Initial Dali XSD

### PR DESCRIPTION
**Pull request by kenrowland**
Convert Dali to new format
Change XSDs to use new default custom namespace attributes
Add hpcc:hiddenIf custom namespace attribute

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [ ] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by kenrowland** [comment]
@rpastrana Please review. 
Note this PR contains the change for the new default specification. All XSDs had to be changed along with the use in Dali

**Comment by rpastrana** [comment]
@kenrowland this needs to be rebased

**Comment by kenrowland** [comment]
@rpastrana This PR is based on previous PRs and one active PR. I would only be able to eliminate 1 commit at this time because the other PR is not merged. I would prefer to keep the changes isolated so I can manage them better. The PR will of course be fully rebased prior to a merge request. Is it OK if you only look at the last commit? It represents all the changes for the subject of the PR

File: configuration/config2/XSDSchemaParser.cpp
Comment (rpastrana) [inline configuration/config2/XSDSchemaParser.cpp]: what is actually being done here? 
we know one of hiddenIf or visibleIf is empty, but why are you appending hiddenif and visibleIf ?

File: configuration/config2/XSDSchemaParser.cpp
Comment (rpastrana) [inline configuration/config2/XSDSchemaParser.cpp]: why is this function called setCodeDefault, it should be setPreset

File: configuration/config2/XSDSchemaParser.cpp
Comment (rpastrana) [inline configuration/config2/XSDSchemaParser.cpp]: correct the method name

File: initfiles/componentfiles/configschema/xsd/dafilesrv.xsd
Comment (rpastrana) [inline initfiles/componentfiles/configschema/xsd/dafilesrv.xsd]: make sure the component owner reviews the xsd, I am not reviewing the xsd

**Review by rpastrana — CHANGES_REQUESTED** [review]
@kenrowland  make sure the XSDs are reviewed by component experts

File: configuration/config2/XSDSchemaParser.cpp
Comment (kenrowland) [reply configuration/config2/XSDSchemaParser.cpp]: The presetValue and forcedConfigValue XSD attributes are for the XSDs so that they attributes are clear as to what they mean. The ESP interface still uses a default in code naming convention, therefore I kept the naming convention throughout the code so that the above was limited to the XSD and parsing. All other code remained unchanged.

File: configuration/config2/XSDSchemaParser.cpp
Comment (kenrowland) [reply configuration/config2/XSDSchemaParser.cpp]: See previous comment.

File: initfiles/componentfiles/configschema/xsd/dafilesrv.xsd
Comment (kenrowland) [reply initfiles/componentfiles/configschema/xsd/dafilesrv.xsd]: that is the next step. I will open a JIRA for that specifically.

**Review by kenrowland — COMMENTED** [review]
@rpastrana Please see comments

**Comment by kenrowland** [comment]
@rpastrana Just a note that the exp.xsd conflict will be resolved once PR 11554 is merged. The PR is dependent on that PR.

**Comment by richardkchapman** [comment]
Please rebase

**Comment by kenrowland** [comment]
@richardkchapman I am waiting for PR 11554 to be merged first as there are dependencies on that PR.

**Comment by kenrowland** [comment]
@rpastrana If there are no more comments, please approve

File: configuration/config2/XSDSchemaParser.cpp
Comment (rpastrana) [reply configuration/config2/XSDSchemaParser.cpp]: this is going to cause maintainability problems, let's rewrite using verbose if else cluases

File: configuration/config2/XSDSchemaParser.cpp
Comment (rpastrana) [reply configuration/config2/XSDSchemaParser.cpp]: I see, let's get all interfaces in synch w/ the new naming convention

**Review by rpastrana — CHANGES_REQUESTED** [review]
@kenrowland a couple of comments

File: configuration/config2/XSDSchemaParser.cpp
Comment (kenrowland) [reply configuration/config2/XSDSchemaParser.cpp]: Changed to use a ternary operator so it's clear which path attribute is being used.

File: configuration/config2/XSDSchemaParser.cpp
Comment (kenrowland) [reply configuration/config2/XSDSchemaParser.cpp]: Agreed. Will change in a later PR to ensure consistency of naming convention

**Review by kenrowland — COMMENTED** [review]
@rpastrana Updated per request. See comments and code change.

**Review by rpastrana — APPROVED** [review]
@kenrowland

**Comment by richardkchapman** [comment]
Needs rebasing/squashing

**Comment by kenrowland** [comment]
@richardkchapman Please merge

**Pull request merged**
